app/services/sale_creator.rb

```diff
@@ -78,12 +78,21 @@ def product_ids
       product_ids = []
       product_ids.concat sale_params[:product_ids]
       if sale_params[:taxon_ids].any?
-        product_ids.concat Spree::ProductsTaxon.where(taxon_id: sale_params[:taxon_ids]).pluck(:product_id)
+        sale_params[:taxon_ids].each do |taxon_id|
+          product_ids.concat nested_products(taxon_id).pluck(:product_id)
+        end
       end
       product_ids
     end
   end
 
+  def nested_products(taxon_id)
+    top_taxon = Spree::Taxon.find(taxon_id)
+    children_taxon_ids = top_taxon.taxonomy.taxons.pluck(:id)
+    all_taxon_ids = [top_taxon.id] + children_taxon_ids
+    Spree::ProductsTaxon.where(taxon_id: all_taxon_ids)
+  end
```

Comment:
aaaah, i just stumbled upon `taxon.self_and_descendants` method so I think you can use it here :D

Reply:
Good find 👍 
We need to leave this PR open and merge just the 3.0 one. When the time comes we'll need to decide how much to bump the gem version as this is a breaking change.

Reply:
Well, I wouldn't say it's a breaking change. It should have been like this all along, so it's just a bugfix :D